src/mrpro/data/_kdata/KData.py

```diff
@@ -109,6 +109,40 @@ def from_file(
             modification_time = datetime.datetime.fromtimestamp(mtime)
 
         acquisitions = [acq for acq in acquisitions if acquisition_filter_criterion(acq)]
+
+        # Select only acquisitions with the desired number of receiver channels/coils.
+        # If the number of coils is not defined use all acquisitions
+        n_coils_available = {acq.data.shape[0] for acq in acquisitions}
+
+        # We need to handle the n_coils parameter of header_overwrites here separately rather than in
+        # KHeader.from_ismrmrd, because the acquisitions have to be selected before we can calculate the header
+        if len(n_coils_available) > 1:
+            if header_overwrites is not None and 'n_coils' in header_overwrites:
+                if isinstance(header_overwrites['n_coils'], int | float):
+                    n_coils = int(header_overwrites['n_coils'])
+                else:
+                    raise ValueError(f'n_coils has to be numeric but is {type(header_overwrites["n_coils"])}')
+            elif (
+                ismrmrd_header.acquisitionSystemInformation is not None
+                and ismrmrd_header.acquisitionSystemInformation.receiverChannels is not None
+            ):
+                n_coils = int(ismrmrd_header.acquisitionSystemInformation.receiverChannels)
+            else:
+                n_coils = int(max(n_coils_available))
+
+            warnings.warn(
+                f'Acquisitions with different number {n_coils_available} of receiver coil elements detected.'
+                'Data with {n_coils} receiver coil elements will be used.',
+                stacklevel=1,
+            )
+
+            # Select acquisition with n_coils receiver coils
+            acquisitions = [acq for acq in acquisitions if has_n_coils(n_coils, acq)]
+
+        # Make sure there are any acquisitions left
```

Comment:
nit-picking: does this comment help to understand what is going on here, considering the clearness of the following two lines ?